### ISSUE TITLE
Number of Action Classes and Action Segments Slightly Different From the Paper.

### ISSUE THREAD

**Issue by guglielmocamporese** [epic-kitchens/epic-kitchens-100-annotations#13]
When I process the action segments from the CSVs (`EPIC_100_train.csv` and `EPIC_100_validation.csv`) I obtain slightly different numbers with respect to the EK-100 paper [[arXiv](https://arxiv.org/pdf/2006.13256.pdf)].

I'm referring to:
- the number of train action segments, 
- the number train action classes, 
- the number validation action classes.

Here I reported the numbers from what I obtain and the paper.

| Source | Partition | Action Segments | Action Classes |
| - | - | - | - |
| Paper [[arXiv](https://arxiv.org/pdf/2006.13256.pdf)] | Train | **67,219** | **3,550** |
| Code below | Train | **67,217** | **3,568**  |

| Source | Partition | Action Segments | Action Classes |
| - | - | - | - |
| Paper [[arXiv](https://arxiv.org/pdf/2006.13256.pdf)] | Validation | 9,668 | **1,344** |
| Code below | Validation | 9,668 | **1,352**  |

Here the code for reproducing the numbers:

```python
import pandas as pd

def get_actions(df):
    """Encode the action as '{v_class}_{n_class}'."""
    return [f'{v}_{n}' for v, n in zip(df['verb_class'], df['noun_class'])]

# Load CSVs and get actions
actions_train = get_actions(pd.read_csv('EPIC_100_train.csv'))
actions_validation = get_actions(pd.read_csv('EPIC_100_validation.csv'))

# Output
train_action_seg = len(actions_train) # -> 67217
train_action_classes = len(set(actions_train)) # -> 3568
validation_action_seg = len(actions_validation) # -> 9668
validation_action_classes = len(set(actions_validation)) # -> 1352
```

- Is this gap expected? 
- What is the correct number of action classes? 
- Could you share a csv where you identify the actions (a = [v_class, n_class])?

Best,

Guglielmo

**Comment by mwray** [comment]
Hi Guglielmo,

Thanks for posting this, there is indeed a discrepancy between the numbers which stemmed from an internal pre-release version of the dataset which weren't updated for the latest arXiv release by mistake.

The numbers that you posted are indeed correct, i.e.

## Training Set
\# Action Segments: 67,217
\# Action Classes: 3,568

## Validation Set
\# Action Classes: 1,352

Mike
